### PR TITLE
feat(cup/argo): add new argo example project

### PR DESCRIPTION
**Pull request by GeorgeMac**
This PR creates a new labs experiment for running Cup and Argo together in `kind`.
The goal is to demonstrate how `cup` can be combined with Argo to streamline the deployment configuration experience.
Quickly read, create, edit and delete deployment configurations via the cupd api and the `cup` CLI.
All changes flow through a pull request flow against a pre-baked Gitea configuration repo.

### CODE REVIEW

File: cup/argo/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+Cup + ArgoCD
+------------
+
+This labs section explore configuring an end to end CD pipeline with Argo and Cup.
+
+![Cup with ArgoCD Diagram](./diagram.svg)
+
+The project deploys a simple application, via Argo, which responds with a message from an environment variable.
```
Comment (markphelps) [inline cup/argo/README.md:8]:
```suggestion
The project deploys a simple application via Argo, which responds with a message from an environment variable.
```

**Review by markphelps — CHANGES_REQUESTED** [review]
some minor wording changes

File: cup/argo/diagram.d2 [hunk 2]
```diff
@@ -0,0 +1,31 @@
+direction: right
```
[the remark this answers is not in the document]
Comment (GeorgeMac) [reply cup/argo/diagram.d2:1]: love a bit of d2

**Review by markphelps — APPROVED** [review]
☕️

**Review by yquansah — APPROVED** [review]
Really clean 🧼

**Pull request merged**